### DISCUSSION TITLE
librechat.yaml/ directory is created

### DISCUSSION THREAD

**Discussion by s2terminal** [danny-avila/LibreChat#2225] [Feature Requests & Suggestions]
I used deploy-compose to set up LibreChat. I want to edit librechat.yaml for configuration. But I found the `librechat.yaml/` directory. Is this the intended behavior?

```
$ ls -pa librechat.yaml/
./  ../
```

I think it would be better to modify the volumes path in [compose yaml](https://github.com/danny-avila/LibreChat/blob/b5d25f5e4f8bafe8e0e58228006218af11182375/deploy-compose.yml#L26), and use [long syntax](https://docs.docker.com/compose/compose-file/compose-file-v3/#long-syntax-3) so that the directory is not created implicitly.

**Comment by danny-avila** [comment]
> I think it would be better to modify the volumes path in [compose yaml](https://github.com/danny-avila/LibreChat/blob/b5d25f5e4f8bafe8e0e58228006218af11182375/deploy-compose.yml#L26), and use [long syntax](https://docs.docker.com/compose/compose-file/compose-file-v3/#long-syntax-3) so that the directory is not created implicitly.

I agree, feel free to submit a PR if it works for you

**Reply by s2terminal** [reply]
I saw that `deploy-compose.yaml` has been modified.
When I deleted the `librechat.yaml/` directory, the directory was not created and did not up.

```
$ rm -rf librechat.yaml/
$ docker compose -f deploy-compose.yml up
```

```
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu/291d448d08fbcc7ef89ee4a288c57bff0037d534ba9cd285824c46a562d7bb51" to rootfs at "/app/librechat.yaml": mount /run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu/291d448d08fbcc7ef89ee4a288c57bff0037d534ba9cd285824c46a562d7bb51:/app/librechat.yaml (via /proc/self/fd/6), flags: 0x5000: no such file or directory: unknown
Gracefully stopping... (press Ctrl+C again to force)
```


I created the `librechat.yaml` file and built it, and it did up fine.

```
$ echo "version: 1.0.5
cache: true
" > librechat.yaml
```
```
$ docker compose -f deploy-compose.yml up --build
```


Thank you very much.

**Comment by alozta2** [comment]
In case anyone is interested, changing the docker mount approach fixed the same issue for me.
```
    # Mounts librechat.yaml as directory which does not make sense
    volumes:
      - type: bind
        source: ./librechat.yaml
        target: /app/librechat.yaml

    # Mounts librechat.yaml as file
    volumes:
      - ./librechat.yaml:/app/librechat.yaml
```
I am not really sure if this is platform issue or not. But I don't think it's related to LibreChat. I was on wsl session in windows 11 while running docker compose commands if anyone has some ideas why it happened.
